aten/src/ATen/native/cuda/AdaptiveMaxPooling3d.cu

```diff
@@ -314,7 +314,7 @@ TORCH_IMPL_FUNC(adaptive_max_pool3d_out_cuda)
 
   checkAllSameGPU(
       __func__, {output_arg, indices_arg, input_arg});
-  if (input.numel() == 0) {
+  if (output.numel() == 0) {
```

Comment:
To fix forward crash in `adaptive_max_pool3d` when output size = 0.
#### Now
```python
>>> import torch
>>> output_size = 0
>>> input = torch.rand([1, 64, 10, 9, 8], dtype=torch.float32, device='cuda', requires_grad=True)
>>> res = torch.nn.functional.adaptive_max_pool3d(input, output_size)
>>> res
tensor([], device='cuda:0', size=(1, 64, 0, 0, 0),
       grad_fn=<AdaptiveMaxPool3DBackward0>)
```

Reply:
Does `input.numel() == 0` case handled now? I think it has to be `input.numel() == 0 || output.numel() == 0`? 

Also, if also tests passed after this change, then we need to have a test with empty input.

Reply:
> Is input.numel() == 0 case handled now?

Actually no, there were some failures. I included those cases too. Could you please re-check? Thanks.

Reply:
Sorry, I am not sure I understand which case you are talking about.

Reply:
The previous case: `input.numel() == 0`. Could you please check the above lines before making these edits to the files? (These were my first commit so that's not updated here). Sorry about the confusion. I'll combine them as you reviewed them. Thanks!